File: jeonghyeon/13week/로또의 최고 순위와 최저 순위.js
function solution(lottos, win_nums) {
    var answer = []; // 답
    let zero = 0; // 0으로 표시한 숫자 개수
    let num = 0; // 맞는 숫자 개수
    for(let n of lottos) {
        if(n === 0) { // 0으로 표시된 숫자 개수 증가
            zero++; 
        } else if(win_nums.includes(n)) {
            num++; // win_nums에 있는 숫자라면 맞는 숫자 개수 증가
        }
    }
    
    let max = (zero === 0 && num === 0) ? 6: 7 - (zero + num); // 0으로 표시된 것이 없고 다 틀리면 6, 아니면 맞는 개수num과 zero를 다 맞다 생각한 수
    let min = num === 0 ? 6 : 7 - num; // 맞는 개수가 0개면 6, 맞는 개수가 있으면 그 등수
    return [max, min];
}

File: gang/week13/로또의 최고 순위와 최저 순위.java
class Solution {
    public int[] solution(int[] lottos, int[] win_nums) {
        int[] answer = new int[2];
        int zeroCnt = 0;
        int matchCnt = 0;
        for(int i=0; i<6; i++) {
            for(int j=0; j<6; j++) {
                if(lottos[i] == 0) {    // 로또 번호가 0인 경우 zeroCnt 증가
                    zeroCnt++;
                    break;
                }
                else if(lottos[i] == win_nums[j]) { // 로또 번호와 당첨 번호가 같은 경우 matchCnt 증가
                    matchCnt++;
                }
            }
        }

        answer[0] = 7 - (zeroCnt+matchCnt); // 최고 순위는 당첨 횟수와 0인 횟수이고 순위이기 때문애 7에서 빼준다

        if(matchCnt == 0) { // 만약 당점 번호가 없고
            if(zeroCnt == 0) {  // 0인 숫자도 없으면
                answer[0] = 6;  // 최고 순위도 6위
            }
            answer[1] = 6;  // 0인 숫자가 있어도 최저 순위는 6위
        }
        else {
            answer[1] = 7 - matchCnt;   // 당첨 번호가 있으면 최저 순위는 7에서 당첨번호 뺀 수
        }

        return answer;
    }
}